core/src/main/java/cucumber/runtime/formatter/PrettyFormatter.java

```diff
@@ -150,8 +151,18 @@ private void handleTestStepFinished(TestStepFinished event) {
     }
 
     private void handleWrite(WriteEvent event) {
-        out.println(event.text);
-    }
+        out.println();
+        try(BufferedReader lines = new BufferedReader(new StringReader(event.text))) {
+		String line = lines.readLine();
+		while(line != null) {
+			out.println(STEP_SCENARIO_INDENT + line);
+			line = lines.readLine();
+		}
+	} catch (IOException e) {
```

Comment:
The canonical way to write this avoids the repetition of `readLine`. It is a little bit less error prone.

```
            String line;
            while ((line = lines.readLine()) != null) {
                out.println(STEP_SCENARIO_INDENT + line);
            }
```

Aside from that looks good. Don't forget to indent your code properly.